polyfill/lib/ecmascript.mjs

```diff
@@ -60,6 +60,27 @@ const NS_MAX = bigInt(86400).multiply(1e17);
 const YEAR_MIN = -271821;
 const YEAR_MAX = 275760;
 const BEFORE_FIRST_DST = bigInt(-388152).multiply(1e13); // 1847-01-01T00:00:00Z
+const BUILTIN_FIELDS = new Set([
+  'year',
+  'month',
+  'day',
+  'hour',
+  'minute',
+  'second',
+  'millisecond',
+  'microsecond',
+  'nanosecond',
+  'years',
+  'months',
+  'weeks',
+  'days',
+  'hours',
+  'minutes',
+  'seconds',
+  'milliseconds',
+  'microseconds',
+  'nanoseconds'
+]);
```

Comment:
Edge case for followup: this doesn't match the spec if `calendar.fields` returns one of the builtin fields that don't apply to the object in question.

Reply:
I added this question to #1047 for followup, as it seems very similar.